sympy/solvers/inequalities.py

```diff
@@ -1004,3 +1004,242 @@ def reduce_inequalities(inequalities, symbols=[]):
+
+    >>> from sympy.solvers.inequalities import _pick_var
+    >>> from sympy.abc import x, y, z
+
+    >>> eq1 = 2*x - 3*y + z + 1
+    >>> eq2 = x - y + 2*z - 2
+    >>> eq3 = x + y + 3*z + 4
+    >>> eq4 = x - z
+
+    >>> inequalities = [eq1, eq2, eq3, eq4]
+    >>> _pick_var(inequalities)
+    x
+    """
+    for eq in inequalities:  # should already be in canonical order
+        for symb in ordered(eq.free_symbols):  # make selection canonical
+            return symb
+
+
+def _fourier_motzkin_extension(inequalities):
+    """Extension of the Fourier-Motzkin algorithm to the case where
+    inequalities do not contain variables that have at least two
+    coefficients with opposite sign.
+
+    Examples
+    ========
+
+    >>> from sympy.solvers.inequalities import _fourier_motzkin_extension
+    >>> from sympy.abc import x, y, z
+
+    >>> eq1 = 2*x - 3*y + z + 1
+    >>> eq2 = x - y + 2*z - 2
+    >>> eq3 = x - y + 3*z + 4
+    >>> eq4 = x + z
+
+    >>> d = _fourier_motzkin_extension([eq1, eq2, eq3, eq4])
+    >>> assert set(d) == {x}
+    >>> d[x]
+    (oo > x, x > Max(-z, y - 3*z - 4, y - 2*z + 2, 3*y/2 - z/2 - 1/2))
+    >>> _fourier_motzkin_extension([x - 3, 5 - x])
+    {x: (5 > x, x > 3)}
+    """
+
+    res = {}
+    pivot = _pick_var(inequalities)
+    while pivot and inequalities:
+        mins, maxs, extra = _split_min_max(inequalities, pivot)
+        res[pivot] = (mins > pivot, pivot > maxs)
+        inequalities = extra
+        pivot = _pick_var(inequalities)
+    return res
+
+
+def solve_linear_inequalities(inequalities):
+    """Solve a system of linear inequalities
+
+    Parameters
+    ==========
+
+    inequalities: list of sympy equations
+        The system of inequalities to solve. All equations in the list
+        are assumed to be linear and greater than 0. The system must
+        be expressed as follows:
+
+        2x - 3y +  z + 1 > 0
+        x  -  y + 2z - 2 > 0
+        x  +  y + 3z + 4 > 0
+        x       -  z     > 0
+
+    Examples
+    ========
+
+    >>> from sympy.solvers.inequalities import solve_linear_inequalities
+    >>> from sympy.abc import x, y, z
+
+    >>> eq1 = 2*x - 3*y + z + 1
+    >>> eq2 = x - y + 2*z - 2
+    >>> eq3 = x + y + 3*z + 4
+    >>> eq4 = x - z
+
+    >>> d = solve_linear_inequalities([eq1, eq2, eq3, eq4])
+    >>> assert set(d) == set([x, y, z])
+    >>> d[x]
+    (oo > x) & (x > -2/7)
+    >>> d[y]
+    (y > -4*x - 4) & (Min(x + 1/3, 3*x - 2) > y)
+    >>> d[z]
+    (x > z) & (z > Max(-2*x + 3*y - 1, -x/2 + y/2 + 1, -x/3 - y/3 - 4/3))
+
+    Explanation
+    ===========
+
+    x = 2 is valid because: oo > 2 > -2/7
+    z = 1 is valid because: x > 1 > Max(-x/2 - 13/10, -2*x/5 - 2/5)
+    y = -1 is valid because: Min(2*x/3 + z/3 + 1/3, x + 2*z - 2) > -1 > -x - 3*z - 4
+    """
+    eqs = list(ordered(inequalities))
+    eqs, res1 = _fourier_motzkin(eqs)
+    res2 = _fourier_motzkin_extension(eqs)
+    return {k: And(*v) for k, v in {**res1, **res2}.items()}
```

Comment:
```suggestion
    >>> d = solve_linear_inequalities([eq1, eq2, eq3, eq4])
    >>> assert set(d) == set([x, y, z])
    >>> d[x]
    (oo > x, x > -2/7)
    >>> d[y]
    (Min(x + 1/3, 3*x - 2) > y, y > -4*x - 4)
    >>> d[z]
    (x > z, z > Max(-2*x + 3*y - 1, -x/2 + y/2 + 1, -x/3 - y/3 - 4/3))

    Explanation
    ===========

    x = 2 is valid because: oo > 2 > -2/7
    z = 1 is valid because: x > 1 > Max(-x/2 - 13/10, -2*x/5 - 2/5)
    y = -1 is valid because: Min(2*x/3 + z/3 + 1/3, x + 2*z - 2) > -1 > -x - 3*z - 4
    """
    eqs = list(ordered(inequalities))
    eqs, res1 = _fourier_motzkin(eqs)
    res2 = _fourier_motzkin_extension(eqs)
    return {**res1, **res2}
```